modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIMANAGER4373BrokenAPIInStoreTestCase.java

```diff
@@ -0,0 +1,162 @@
+import org.wso2.carbon.user.mgt.stub.UserAdminUserAdminException;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.net.URL;
+import java.rmi.RemoteException;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * None of the apis are visible in API Store when permissions for one API has changed.
+ * This is test class tests the fix for this issue
+ *
+ * @see <a href="https://wso2.org/jira/browse/APIMANAGER-4373">APIMANAGER-4373</a>
+ */
+public class APIMANAGER4373BrokenAPIInStoreTestCase extends APIMIntegrationBaseTest {
+    private final String USER_PASSWORD = "123123";
+    private final String PERMISSION_LOGIN = "/permission/admin/login";
+    private final String PERMISSION_API_SUBSCRIBE = "/permission/admin/manage/api/subscribe";
+    private final String FIRST_USER = "APIMANAGER4373_user";
+    private final String FIRST_ROLE = "APIMANAGER4373_role1";
+    private final String SECOND_ROLE = "APIMANAGER4373_role2";
+    private final String APP_NAME = "APIMANAGER4373";
+    private final String BROKEN_API = "brokenAPI";
+    private final String HEALTHY_API = "healthyAPI";
+    private final String EP_URL = "http://gdata.youtube.com/feeds/api/standardfeeds";
+    private final String API_VERSION = "1.0.0";
+    private String contextUsername = "admin";
+    private String contextUserPassword = "admin";
+    private UserManagementClient userManagementClient;
+    private APIPublisherRestClient apiPublisher;
+    private APIStoreRestClient apiStoreRestClient;
+    private String publisherURLHttp;
+    private String storeURLHttp;
+
+    @BeforeMethod
+    public void init() {
+        try {
+            super.init();
+            publisherURLHttp = publisherUrls.getWebAppURLHttp();
+            storeURLHttp = storeUrls.getWebAppURLHttp();
+            apiPublisher = new APIPublisherRestClient(publisherURLHttp);
+            apiStoreRestClient = new APIStoreRestClient(storeURLHttp);
+            contextUsername = publisherContext.getContextTenant().getContextUser().getUserName();
+            contextUserPassword = publisherContext.getContextTenant().getContextUser().getPassword();
+            userManagementClient = new UserManagementClient(publisherContext.getContextUrls().getBackEndUrl(),
+                    contextUsername, contextUserPassword);
+
+            userManagementClient
+                    .addRole(FIRST_ROLE, new String[] {}, new String[] { PERMISSION_LOGIN, PERMISSION_API_SUBSCRIBE });
+            userManagementClient
+                    .addRole(SECOND_ROLE, new String[] {}, new String[] { PERMISSION_LOGIN, PERMISSION_API_SUBSCRIBE });
+            userManagementClient.addUser(FIRST_USER, USER_PASSWORD, new String[] { FIRST_ROLE }, FIRST_USER);
+        } catch (APIManagerIntegrationTestException e) {
+            assertTrue(false, "Error occurred while initializing testcase.");
+        } catch (RemoteException e) {
+            assertTrue(false, "Error occurred while adding new users.");
```

Comment:
No need to catch the exceptions and have asserts in here. Just throw the exceptions